search.go

```diff
@@ -105,26 +111,49 @@ type FacetRequest struct {
 }
 
 func (fr *FacetRequest) Validate() error {
-	if len(fr.NumericRanges) > 0 && len(fr.DateTimeRanges) > 0 {
+	nrCount := len(fr.NumericRanges)
+	drCount := len(fr.DateTimeRanges)
+	if nrCount > 0 && drCount > 0 {
 		return fmt.Errorf("facet can only conain numeric ranges or date ranges, not both")
 	}
-
-	nrNames := map[string]interface{}{}
-	for _, nr := range fr.NumericRanges {
-		if _, ok := nrNames[nr.Name]; ok {
-			return fmt.Errorf("numeric ranges contains duplicate name '%s'", nr.Name)
+	rangeSet := false
+	if nrCount > 0 {
+		nrNames := map[string]interface{}{}
+		for _, nr := range fr.NumericRanges {
+			if _, ok := nrNames[nr.Name]; ok {
+				return fmt.Errorf("numeric ranges contains duplicate name '%s'", nr.Name)
+			}
+			nrNames[nr.Name] = struct{}{}
+			if !rangeSet && (nr.Min != nil || nr.Max != nil) {
+				rangeSet = true
+			}
+		}
+		if !rangeSet {
+			return fmt.Errorf("numeric range query must specify either min, max or both")
```

Comment:
I think this logic can miss cases where the ranges are missing.  For example, consider a case where there are 2 numeric ranges.  Initially rangeSet is false, we loop over the numeric ranges.  Let's say the first range does have a minimum set.  Now rangeSet is set to true.  Now we look at the second range, and let imagine it doesn't have either range set.  But, we won't even check that because !rangeSet is false now.  After exiting the loop, since !rangeSet is false, no error message is returned... but one of the ranges was improperly specified.  And I think the same problem exists for dates below.

Of course I could be reading it wrong, but to @steveyen point, additional tests would help make it clear that we verified these cases.